### ISSUE TITLE
Improve suggestion on annotation fields

### ISSUE THREAD

**Issue by raulvaldoleirospowerdot** [redhat-developer/vscode-java#3583]
When I'm adding properties to annotation the suggestion doesn't help, for instance, in the following image the property forRemoval is a boolean and the system doesn't suggest true or false.

![image](https://github.com/redhat-developer/vscode-java/assets/152172293/16dfa767-d0fd-4691-a324-637eed14d85a)

##### Environment
- Operating System: Mac OS
- JDK version: 21
- Visual Studio Code version: Version: 1.88.1 (Universal)
- Java extension version: 1.29.0

**Comment by rgrunber** [comment]
I think this might be a case of not suggesting basic primitives since we do prioritize values that make sense in that context.

**boolean**
![image](https://github.com/redhat-developer/vscode-java/assets/1417342/29eedf1b-6c33-444e-af38-c6b3dbf34783)

**String**
![image](https://github.com/redhat-developer/vscode-java/assets/1417342/51368dfc-77da-4d66-a9f0-b645990f5c2a)

**Comment by mamilic** [comment]
@rgrunber , is that intellisence or github copilot suggestions?

**Comment by rgrunber** [comment]
It's just plain intellisense. Specifically, the language server for vscode-java sorts the entries, and prioritizes those that are relevant in that context (ie. the the type/return type matches the annotation parameter value's type.

**Comment by rgrunber** [comment]
Right. Currently they're sorted in alphabetical order when their relevance is the same. Maybe we could reduce the relevance of the methods inherited from `java.lang.Object`. Or maybe https://github.com/eclipse-jdtls/eclipse.jdt.ls/issues/2484 would fix this. The more frequently used methods would get displayed higher.
